dashboard/app/helpers/users_helper.rb

```diff
@@ -11,7 +11,7 @@ def check_and_apply_oauth_takeover(user)
     if session['clever_link_flag'].present? && session['clever_takeover_id'].present?
       uid = session['clever_takeover_id']
       # TODO: validate that we're not destroying an active account?
-      existing_account = User.where(uid: uid, provider: session['clever_link_flag']).first
+      existing_account = User.find_by_credential(type: session['clever_link_flag'], id: uid)
```

Comment:
We use `session['clever_link_flag']` a couple of times in this function, it'd be nice to assign it to a `provider` variable up top like we do for `uid`.